mods/ra/maps/soviet-06a/soviet06a.lua

```diff
@@ -0,0 +1,154 @@
+	if not cameraBarrierTrigger and a.Owner == player then
+		cameraBarrierTrigger = true
+		local cameraBarrier = Actor.Create("camera", true, { Owner = player, Location = CameraBarrier.Location })
+		Trigger.AfterDelay(DateTime.Seconds(15), function()
+			cameraBarrier.Kill()
+		end)
+	end
+end)
+
+Trigger.OnEnteredFootprint(CameraBaseTrigger, function(a, id)
+	if not cameraBaseTrigger and a.Owner == player then
+		cameraBaseTrigger = true
+		local cameraBase1 = Actor.Create("camera", true, { Owner = player, Location = CameraBase1.Location })
+		local cameraBase2 = Actor.Create("camera", true, { Owner = player, Location = CameraBase2.Location })
+		local cameraBase3 = Actor.Create("camera", true, { Owner = player, Location = CameraBase3.Location })
+		local cameraBase4 = Actor.Create("camera", true, { Owner = player, Location = CameraBase4.Location })
+		Trigger.AfterDelay(DateTime.Minutes(1), function()
+			cameraBase1.Kill()
+			cameraBase2.Kill()
+			cameraBase3.Kill()
+			cameraBase4.Kill()
+		end)
+	end
+end)
+
+Trigger.OnAllKilled(Trucks, function()
+	if not controlCenterTrigger then
+		enemy.MarkCompletedObjective(alliedObjective)
+	end
+end)
+
+Trigger.OnKilled(Apwr, function(building)
+	BaseBuildings[1][4] = false
+end)
+
+Trigger.OnKilled(Barr, function(building)
+	BaseBuildings[2][4] = false
+end)
+
+Trigger.OnKilled(Proc, function(building)
+	BaseBuildings[3][4] = false
+end)
+
+Trigger.OnKilled(Weap, function(building)
+	BaseBuildings[4][4] = false
+end)
+
+Trigger.OnKilled(Apwr2, function(building)
+	BaseBuildings[5][4] = false
+end)
+
+-- Activate the AI once the player deployed the Mcv
+Trigger.OnRemovedFromWorld(Mcv, function()
+	if not mcvDeployed then
+		mcvDeployed = true
+		BuildBase()
+		Trigger.AfterDelay(DateTime.Minutes(1), ProduceInfantry)
+		Trigger.AfterDelay(DateTime.Minutes(2), ProduceArmor)
+		Trigger.AfterDelay(DateTime.Minutes(2), function()
+			Utils.Do(BaseAttackers, function(actor)
+				if not actor.IsDead then
+					Trigger.OnIdle(actor, actor.Hunt)
+				end
+			end)
+		end)
+	end
+end)
+
+WorldLoaded = function()
+	player = Player.GetPlayer("USSR")
+	enemy = Player.GetPlayer("Greece")
+	Camera.Position = CameraStart.CenterPosition
+	if not Mcv.IsDead then
+		Mcv.Move(McvWaypoint.Location)
+	end
+	if not Harvester.IsDead then
+		Harvester.FindResources()
+	end
+	Utils.Do(IntroAttackers, function(actor)
+		if not actor.IsDead then
+			Trigger.OnIdle(actor, actor.Hunt)
+		end
+	end)
+	Utils.Do(Map.NamedActors, function(actor)
+		if actor.Owner == enemy and actor.HasProperty("StartBuildingRepairs") then
+			Trigger.OnDamaged(actor, function(building)
+				if building.Owner == enemy and building.Health < 3/4 * building.MaxHealth then
+					building.StartBuildingRepairs()
+				end
+			end)
+		end
+	end)
+	Reinforcements.ReinforceWithTransport(player, "apc", SovietReinforcements1, SovietReinforcements1Waypoints, nil, function(apc, team)
+		table.insert(team, apc)
```

Comment:
I've reused it from gdi04a, maybe you could tell me what it's used for in there?